devserver.py

```diff
@@ -0,0 +1,56 @@
+import http.server
+from os.path import exists
+from os import makedirs
+import socketserver
+import urllib.request
+import webbrowser
+
+PORT = 8000
+PROD_URL = 'https://age4builder.com'
+DIRECTORY = './public'
+IMG_DIRECTORY = f'{DIRECTORY}/img'
+
+Handler = http.server.SimpleHTTPRequestHandler
+
+if not exists(IMG_DIRECTORY):
+    makedirs(IMG_DIRECTORY)
+
+
+def is_dev_image(path: str) -> bool:
+    return '/img/' in path or '.ico' in path
+
+
+def local_dev_url_to_prod_url(path: str) -> str:
+    return f'{PROD_URL}{path}'
+
+
+class DevHandler(Handler):
+    """
+    An HTTP  Handler that serves files from DIRECTORY and fetches images from PROD_URL.
+    """
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, directory=DIRECTORY, **kwargs)
+
+    def do_GET(self) -> None:
+        print(f'GET {self.path}')
+
+        dev_file_path = f'{DIRECTORY}{self.path}'
+        if not exists(dev_file_path) and is_dev_image(self.path):
+            img_url_to_fetch = local_dev_url_to_prod_url(self.path)
+            img_path_to_write = dev_file_path
+
+            print(
+                f'Fetching image from prod URL: {img_url_to_fetch} -> {img_path_to_write}')
+
+            urllib.request.urlretrieve(
+                img_url_to_fetch, img_path_to_write)
+
+        return super().do_GET()
+
+
+with socketserver.TCPServer(("", PORT), DevHandler) as httpd:
+    print("serving at port", PORT)
+    local_url = f'http://localhost:{PORT}'
+    webbrowser.open(local_url, new=2)
```

Comment:
There is a good chance that I am doing something wrong, so take this with a grain of salt, but this line hangs for me.

In order to get things displaying on localhost:8000 I need to comment this out and navigate there in a browser manually

Reply:
Just wanted to follow up with this and say the code worked out of the box for me. Python 3.11.2, Firefox 110.0.1 on Windows 11.

Reply:
Im glad!